### ISSUE TITLE
2015.8.9 minion crashes on startup when using a minion config generated by salt-cloud

### ISSUE THREAD

**Issue by UtahDave** [saltstack/salt#33384]
### Description of Issue/Question
I have a minion that was created by an older version of salt-cloud. I upgraded to 2015.8.9 from repo.saltstack.com and everything seemed to be good until I ran this command:

```
[root@myserver me]# salt-call test.version
[ERROR   ] 'tuple' object has no attribute '__name__'
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/utils/parsers.py", line 169, in parse_args
    process_option_func()
  File "/usr/lib/python2.7/site-packages/salt/utils/parsers.py", line 451, in process_config_dir
    self.config.update(self.setup_config())
  File "/usr/lib/python2.7/site-packages/salt/utils/parsers.py", line 2276, in setup_config
    cache_minion_id=True)
  File "/usr/lib/python2.7/site-packages/salt/config.py", line 1669, in minion_config
    _validate_opts(opts)
  File "/usr/lib/python2.7/site-packages/salt/config.py", line 1406, in _validate_opts
    format_multi_opt(VALID_OPTS[key].__name__))
AttributeError: 'tuple' object has no attribute '__name__'
Usage: salt-call [options] <function> [arguments]

salt-call: error: Error while processing <bound method SaltCall.process_config_dir of <salt.cli.call.SaltCall object at 0x1b70b50>>: Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/utils/parsers.py", line 169, in parse_args
    process_option_func()
  File "/usr/lib/python2.7/site-packages/salt/utils/parsers.py", line 451, in process_config_dir
    self.config.update(self.setup_config())
  File "/usr/lib/python2.7/site-packages/salt/utils/parsers.py", line 2276, in setup_config
    cache_minion_id=True)
  File "/usr/lib/python2.7/site-packages/salt/config.py", line 1669, in minion_config
    _validate_opts(opts)
  File "/usr/lib/python2.7/site-packages/salt/config.py", line 1406, in _validate_opts
    format_multi_opt(VALID_OPTS[key].__name__))
AttributeError: 'tuple' object has no attribute '__name__'
```
I thought maybe it had something to do with the generated minion config that salt-cloud produces so I swapped out a fresh minion config with only the `master` option that I needed and then everything seems to be working just fine.

### Setup
1.  Install salt using salt-cloud
2.  Upgrade salt-minion

### Steps to Reproduce Issue
run `salt-call test.ping`

### Versions Report
```
[root@pkgrepo-backend01 salt]# salt-minion --versions-report
Salt Version:
           Salt: 2015.8.9
 
Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: 0.21.1
           Mako: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
         Python: 2.7.5 (default, Nov 20 2015, 02:00:19)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed
 
System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 3.10.0-123.8.1.el7.x86_64
         system: CentOS Linux 7.2.1511 Core
```

**Comment by tyhunt99** [comment]
I am seeing a similar issue trying to launch new minions. My master is currently still on 2015.8.7 as I know this version works for me.
Is there a way to specify the version the new salt minion (launched with salt-cloud) should run with?

**Comment by rallytime** [comment]
@tyhunt99 You can use the `script_args` option in your cloud profile to specify a git tag. So if you wanted to use the same version as your master, you'd need `script_args: git v2015.8.7`.

**Comment by tyhunt99** [comment]
@rallytime great thank you!

**Comment by Ch3LL** [comment]
This has been fixed by #33386. I'm going to go ahead and close this since this has been verified as a fix for this issue.
